compiler/src/dotty/tools/dotc/core/Decorators.scala

```diff
@@ -234,6 +234,22 @@ object Decorators {
     def nestedExists(p: T => Boolean): Boolean = xss match
       case xs :: xss1 => xs.exists(p) || xss1.nestedExists(p)
       case nil => false
+    def nestedZipExists(yss: List[List[U]])(f: (T, U) => Boolean): Boolean =
+      @tailrec def zipExists(p1: List[T], p2: List[U]): Boolean =
```

Comment:
I think we can do without `nestedZipExists` for the time being. I'd remove it in the interest of keeping the codebase shorter.